src/addrman.cpp

```diff
@@ -521,3 +533,84 @@ void CAddrMan::SetServices_(const CService& addr, ServiceFlags nServices)
 int CAddrMan::RandomInt(int nMax){
     return GetRandInt(nMax);
 }
+
+void CAddrMan::ResolveCollisions_()
+{
+    for (std::set<int>::iterator it = m_tried_collisions.begin(); it != m_tried_collisions.end();) {
+        int id_new = *it;
+
+        bool erase_collision = false;
```

Comment:
It seems like you can reduce the lines below by defaulting to `true` here, as several cases simply turn it on. Unless it is too complicated to swap.

Reply:
I appreciate this comment, but I'm concerned it would be more confusing to read if I shortened it.